NDTensors/src/lib/Sectors/test/test_category_product.jl

```diff
@@ -52,7 +40,7 @@ using Test: @inferred, @test, @testset, @test_broken, @test_throws
     @test categories(s)[1] == U1(3)
     @test categories(s)[2] == SU2(1//2)
     @test categories(s)[3] == Fib("τ")
-    @test (@inferred trivial(typeof((s)))) == sector(U1(0), SU2(0), Fib("1"))
+    @test trivial(s) == sector(U1(0), SU2(0), Fib("1"))    # need julia 1.10
```

Comment:
```suggestion
    @test trivial(s) == sector(U1(0), SU2(0), Fib("1"))    # need julia 1.10 for type stability
```
Note that a strategy for designing these kinds of tests that only pass on certain Julia versions is to define our own macro, maybe `@inferred_latest`, that only tests if it is inferred on the latest released Julia version (i.e. 1.10).

For example:
```julia
using Test: @inferred
macro inferred_latest(ex)
  if VERSION < v"1.10"
    return ex
  end
  return :(@inferred $ex)
end
```
On Julia 1.10 and above:
```julia
julia> versioninfo()
Julia Version 1.10.3
Commit 0b4590a5507 (2024-04-30 10:59 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 10 × Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)

julia> f(x) = x < 0.5 ? 1 : 1.0
f (generic function with 1 method)

julia> @inferred f(2)
ERROR: return type Float64 does not match inferred return type Union{Float64, Int64}
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] top-level scope
   @ REPL[4]:1
```
while on older versions:
```julia
julia> versioninfo()
Julia Version 1.9.4
Commit 8e5136fa297 (2023-11-14 08:46 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 10 × Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores

julia> f(x) = x < 0.5 ? 1 : 1.0
f (generic function with 1 method)

julia> @inferred_latest f(2)
1.0
```